src/verify/verify.test.ts

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright 2022 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { mocked } from "jest-mock";
+import {
+  getVerifiableCredentialApiConfiguration,
+  isVerifiableCredential,
+} from "../common/common";
+import isValidVc from "./verify";
+import fallbackFetch from "../fetcher";
+
+jest.mock("../common/common");
+jest.mock("../fetcher");
+
+describe("isValidVc", () => {
+  const MOCK_VC = {
+    "@context": [
+      "https://www.w3.org/2018/credentials/v1",
+      "https://consent.pod.inrupt.com/credentials/v1",
+    ],
+    id: "https://example.com/id",
+    issuer: "https://example.com/issuer",
+    type: ["VerifiableCredential", "SolidCredential", "SolidAccessGrant"],
+    issuanceDate: "2021-05-26T16:40:03",
+    expirationDate: "2021-06-09T16:40:03",
+    credentialSubject: {
+      id: "https://pod.inrupt.com/alice/profile/card#me",
+      providedConsent: {
+        mode: ["http://www.w3.org/ns/auth/acl#Read"],
+        hasStatus: "ConsentStatusExplicitlyGiven",
+        forPersonalData: "https://pod.inrupt.com/alice/private/data",
+        forPurpose: "https://example.com/SomeSpecificPurpose",
+        isProvidedToPerson: "https://pod.inrupt.com/bob/profile/card#me",
+      },
+    },
+    proof: {
+      created: "2021-05-26T16:40:03.009Z",
+      proofPurpose: "assertionMethod",
+      proofValue: "eqp8h_kL1DwJCpn65z-d1Arnysx6b11...jb8j0MxUCc1uDQ",
+      type: "Ed25519Signature2020",
+      verificationMethod: "https://consent.pod.inrupt.com/key/396f686b",
+    },
+  };
+  const MOCK_VERIFY_ENDPOINT = "https://consent.example.com";
+  const MOCK_VERIFY_RESPONSE = { checks: [], warning: [], errors: [] };
+
+  it("falls back to the embedded fetch if none is provided", async () => {
+    const mockedFetch = jest.requireMock("../fetcher") as jest.Mocked<{
+      default: typeof fetch;
+    }>;
```

Comment:
It depends: in this function, the `../fetcher` module is mocked, so that the fallback is tested. For most of the other tests (if not all), a regular mock function is created and passed as the optional fetch. These are two separate construct, and the one pointed out here, which is the one which would make sense to be lifted in the upper scope, is actually not repeated, so I don't think changes are necessary.